src/Temporalio/Worker/WorkflowInstance.cs

```diff
@@ -2453,6 +2491,106 @@ public override Task<ChildWorkflowHandle<TWorkflow, TResult>> StartChildWorkflow
                 return handleSource.Task;
             }
 
+            /// <inheritdoc/>
+            public override Task<NexusOperationHandle<TResult>> StartNexusOperationAsync<TResult>(
+                StartNexusOperationInput input)
+            {
+                var token = input.Options.CancellationToken ?? instance.CancellationToken;
+                // We do not even want to schedule if the cancellation token is already cancelled.
+                // We choose to use cancelled failure instead of wrapping in Nexus failure which is
+                // similar to what Java and TypeScript do, with the accepted tradeoff that it makes
+                // catch clauses more difficult (hence the presence of
+                // TemporalException.IsCanceledException helper).
+                if (token.IsCancellationRequested)
+                {
+                    return Task.FromException<NexusOperationHandle<TResult>>(
+                        new CanceledFailureException("Nexus operation cancelled before scheduled"));
+                }
+
+                var seq = ++instance.nexusOperationCounter;
+                var cmd = new ScheduleNexusOperation()
+                {
+                    Seq = seq,
+                    Endpoint = input.ClientOptions.Endpoint,
+                    Service = input.Service,
+                    Operation = input.OperationName,
+                    Input = input.Arg == null ? null : instance.PayloadConverter.ToPayload(input.Arg),
```

Comment:
to confirm this `PayloadConverter` does not have the workflow context attached correct

Reply:
It does as do all workflow calls. Based on off-PR discussions ,I have opened a separate issue at #523 to solve this generally (it is not a Nexus-specific concern).

Reply:
I think we should fix this for Nexus to not make the problem worse